src/Nethermind/Nethermind.Merge.Plugin/Handlers/V1/NewPayloadV1Handler.cs

```diff
@@ -169,10 +169,10 @@ public async Task<ResultWrapper<PayloadStatusV1>> HandleAsync(ExecutionPayloadV1
 
                 BlockTreeInsertOptions insertOptions = BlockTreeInsertOptions.BeaconBlockInsert;
 
-                if (_blockCacheService.ProcessDestination != null && _blockCacheService.ProcessDestination.Hash == block.ParentHash)
+                if (_beaconPivot.ProcessDestination != null && _beaconPivot.ProcessDestination.Hash == block.ParentHash)
```

Comment:
What if we don't have BeaconPivot here? It could be in two cases:
1. We're doing sync from PoW chain and we have parent (no need to start beacon header sync), but we haven't processed blocks yet.
2. We have a timeout in newPayload(x) and we have next newPayload(x+1) in this case we don't have beacon pivot at all.

So I think if we shouldn't EnsurePivot in this if. WDYT?

Reply:
Ah great... well I'm gonna put that as out of scope.

Reply:
but it could introduce regression in such cases